arraycontext/container/arithmetic.py

```diff
@@ -343,6 +343,17 @@ def {fname}(arg1):
                                 else:
                                     raise ValueError(msg)""")
                     gen(f"return cls({zip_init_args})")
+                if _cls_has_array_context_attr:
+                    gen("if isinstance(arg2,"
+                        "              arg1.array_context.get_array_types()):")
+                    with Indentation(gen):
+                        gen(f"""
+                                return (
+                                    rec_map_array_container(
+                                        lambda a: {op_str.format("a", "arg2")},
+                                        arg1))
+                            """)
+
                 gen(f"""
                 if {bool(outer_bcast_type_names)}:  # optimized away
                     if isinstance(arg2, {tup_str(outer_bcast_type_names)}):
```

Comment:
This should also work (and be tested) for the reverse operators.

Reply:
Added a test in  f0e0587.